mvc/src/main/java/nextstep/mvc/servlet/DispatcherServlet.java

```diff
@@ -0,0 +1,88 @@
+package nextstep.mvc.servlet;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import nextstep.mvc.view.ModelAndView;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class DispatcherServlet extends HttpServlet {
+
+    private static final long serialVersionUID = 1L;
+    private static final Logger log = LoggerFactory.getLogger(DispatcherServlet.class);
+
+    private final List<HandlerMapping> handlerMappings;
+    private final List<HandlerAdapter> handlerAdapters;
+
+    public DispatcherServlet() {
+        this.handlerMappings = new ArrayList<>();
+        this.handlerAdapters = new ArrayList<>();
+    }
+
+    @Override
+    public void init() {
+        for (HandlerMapping handlerMapping : handlerMappings) {
+            initHandlerMapping(handlerMapping);
+        }
+    }
+
+    private void initHandlerMapping(HandlerMapping handlerMapping) {
+        try {
+            handlerMapping.initialize();
+        } catch (InstantiationException | InvocationTargetException | NoSuchMethodException | IllegalAccessException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void addHandlerMapping(HandlerMapping handlerMapping) {
+        handlerMappings.add(handlerMapping);
+    }
+
+    public void addHandlerAdapter(HandlerAdapter handlerAdapter) {
+        handlerAdapters.add(handlerAdapter);
+    }
+
+    @Override
+    protected void service(HttpServletRequest request, HttpServletResponse response) throws ServletException {
+        log.debug("Method : {}, Request URI : {}", request.getMethod(), request.getRequestURI());
+
+        try {
+            final Object handler = getHandler(request);
+            final HandlerAdapter handlerAdapter = getHandlerAdapter(handler);
+
+            if (isContent(handler, handlerAdapter)) return;
+
+            final ModelAndView modelAndView = handlerAdapter.handle(request, response, handler);
+            modelAndView.getView().render(modelAndView.getModel(), request, response);
```

Comment:
디미터의 법칙을 고려한다면 이렇게 동작하도록 구현해 볼 수 있겠네요! ✏️
```suggestion
            modelAndView.render(request, response);
```